syntax = "proto3";

package server.post;

option go_package = "github.com/kwanok/minsim-api/server/post";

service Post {
  rpc NewPost(NewPostRequest) returns (NewPostResponse);
}

message NewPostRequest {
  string title = 1;
  string content = 2;
  string author = 3;
  string subreddit = 4;
  int64 created_at = 5;
  string url = 6;
  string id = 7;
}

message NewPostResponse {
  string id = 1;
}